llama-index-integrations/indices/llama-index-indices-managed-vectara/llama_index/indices/managed/vectara/query.py

```diff
@@ -94,24 +107,31 @@ def with_retriever(self, retriever: VectaraRetriever) -> "VectaraQueryEngine":
 
     def _query(self, query_bundle: QueryBundle) -> RESPONSE_TYPE:
         """Answer a query."""
-        with self.callback_manager.event(
-            CBEventType.QUERY, payload={EventPayload.QUERY_STR: query_bundle.query_str}
-        ) as query_event:
-            kwargs = (
-                {
-                    "summary_response_lang": self._summary_response_lang,
-                    "summary_num_results": self._summary_num_results,
-                    "summary_prompt_name": self._summary_prompt_name,
-                }
-                if self._summary_enabled
-                else {}
+        kwargs = (
+            {
+                "summary_response_lang": self._summary_response_lang,
+                "summary_num_results": self._summary_num_results,
+                "summary_prompt_name": self._summary_prompt_name,
+            }
+            if self._summary_enabled
+            else {}
+        )
+
+        if self._streaming:
+            nodes = self.retrieve(query_bundle)
+
+            query_response = StreamingResponse(
+                response_gen=self._retriever._vectara_stream(query_bundle, chat=False),
+                source_nodes=nodes,
             )
-            nodes, response = self._retriever._vectara_query(query_bundle, **kwargs)
-            query_event.on_end(payload={EventPayload.RESPONSE: response})
-        return Response(response=response, source_nodes=nodes)
+        else:
+            nodes, response, _ = self._retriever._vectara_query(query_bundle, **kwargs)
+            query_response = Response(response=response["text"], source_nodes=nodes)
+
+        return query_response
 
     async def _aquery(self, query_bundle: QueryBundle) -> RESPONSE_TYPE:
-        return self._query(query_bundle)
+        return await self._query(query_bundle)
```

Comment:
I think this won't work, since `self._query()` isn't an async function, we can't use await

Reply:
So just call it directly? or just remove aquery() in this case?

Reply:
Both are viable options 👍

Reply:
Let me remove it then.

Reply:
Actually that doesn't work since _aquery() is requires by the base class.